server/ai_mediaserver.go

```diff
@@ -594,6 +594,7 @@ func (ls *LivepeerServer) StartLiveVideo() http.Handler {
 			if err != nil {
 				clog.Errorf(ctx, "Failed to kick input connection: %s", err)
 			}
+			cleanupLive(ctx, ls.LivepeerNode, streamName)
```

Comment:
Oof, it's hard to follow all these independent logics for stopping the pipeline. The current code is an improvement and probably works for now, but I wish we could have a centralized signal for stopping/cancellation. The idiomatic way would be using that `ctx` as the general pipeline context and then cancelling it from any place that finds a termination signal or error. Then the others can just wait on `ctx.Done()` to cleanup their own state (and this `cleanupLive` could even be called from here on a routine of its own). This would be a bigger refactor tho so just sharing the idea for a future refactor maybe.

Reply:
Yes, totally agree 💯

Reply:
I am a little confused why this addition is necessary here, since we should always call this function after the segmenter closes - prior to this PR, post-segment was the only place the cleanupLive function was called (once for native whip ingest, once for mediamtx ingest).

IIRC the thinking at the time was that we would stop the control channel _after_ we know ingest completes; here, there seems to be a risk of those falling out of sync.